### PR TITLE
Adding GetLayerNames()

### PR DESCRIPTION
**Pull request by Sauraus**
This PR fixes https://github.com/hybridgroup/gocv/issues/388

### CODE REVIEW

**Comment by deadprogram** [comment]
Hello @Sauraus thank you very much for the PR. Could you please add a "smoke test" for it as mentioned here: https://github.com/hybridgroup/gocv/blob/master/CONTRIBUTING.md#how-to-add-a-function-from-opencv-to-gocv

It just needs to have coverage for the newly added function.

Thanks!

**Comment by deadprogram** [comment]
Also, please run `go fmt` on the code to correct the formatting. Thank you.

**Comment by Sauraus** [comment]
@DeadNumbers I ran `go fmt` but did not see any changes to be committed.

**Comment by deadprogram** [comment]
I think it is just matter of adding test coverage in the case of this PR.

**Comment by Sauraus** [comment]
Ok I added the test like for the existing function `GetUnconnectedOutLayers` to the code and pushed it again.

**Comment by deadprogram** [comment]
Hi @Sauraus thanks for adding some test coverage. Unfortunately the tests currently are failing: https://travis-ci.org/hybridgroup/gocv/builds/489752698#L1098

Can you please look into why? I'd love to merge this PR. :)

**Comment by Sauraus** [comment]
Yeah I looking into that unfortunately https://github.com/hybridgroup/gocv/issues/389 is a pain as I am on a Mac.

**Comment by deadprogram** [comment]
Same failure on Appveyor as on Travis: https://ci.appveyor.com/project/deadprogram/gocv/builds/22895911#L302

Any ideas?

**Comment by Sauraus** [comment]
@deadprogram I've not forgotten about this :) just got dragged into something else for the time being.

**Comment by deadprogram** [comment]
All good @Sauraus I know how it goes.

**Comment by deadprogram** [comment]
Hello @Sauraus 

Any chance of bringing this PR back to life? Seems like it is pretty close to completed...

File: dnn_test.go [hunk 1]
```diff
@@ -47,6 +47,11 @@ func TestReadNet(t *testing.T) {
 			t.Errorf("Invalid len output layers in ReadNet test: %d\n", len(ids))
 		}
 
+		lnames := net.GetLayerNames()
+		if len(lnames) != 1 {
```
Comment (deadprogram) [inline dnn_test.go]: For the model being tested here, the number of layers is 142.

I think if this number is changed in the test, the CI will pass and this PR could be merged...

**Comment by deadprogram** [comment]
Trying again to land this PR... it is so close! Just fix the failing test as I commented, and it could be merged.

**Comment by deadprogram** [comment]
I rebased and corrected the test value myself. I then merged both commits into the `dev` branch.

As such, this PR can now be closed as mergedd. Thank you @Sauraus for your contribution.

**Pull request closed without merging**